config/crd/bases/everest.percona.com_backupstorages.yaml

```diff
@@ -49,16 +49,15 @@ spec:
                 description: Region is a region where the bucket is located.
                 type: string
               type:
-                description: Type is a type of backup storage. Currently only S3 is
-                  supported.
+                description: Type is a type of backup storage.
                 enum:
                 - s3
+                - gcs
+                - azure
                 type: string
             required:
             - bucket
             - credentialsSecretName
-            - endpointURL
-            - region
```

Comment:
We're gonna need an update for the validation process on the BE side for that change - check that for the S3 the fields are set. Also we're gonna need to validate the access and write permissions for the GCS and Azure by create/update a backup storage. I couldn't find a ticket for it in Jira, could you please create one so it would be prominent there's work to be done to complete the feature?

Reply:
I agree. I'll look into into backend next. I wanted to make sure the approach in operator is fine first.

Reply:
The approach is fine, yet it would be best to merge it after the backend changes will be there because the spec needs to be updated.

Reply:
Wouldn't it make more sense to switch the order? i.e. merge this, take the schema and put it into backend?

If we were to merge backend first, the schema used in the backend would not be valid before the operator PR is merged.
Also the backend depends on the operator. The operator does not have to worry about the version of the backend.

This seems to me like a more rational approach. Am I missing anything?

Reply:
The merging order does not make much sense but we need to have these updates first. Those changes are related and the PRs should be linked with each other using the 'Related PRs section'. Also, the validation should be present because right now we do not have any validation neither on the operator side nor on the BE side. The validation is also part of the implementation which is missing right now.